### PR TITLE
Add forgot password feature

### LINKED ISSUE #77: Implement login & forgot password pages

### PR DESCRIPTION
**Pull request by htnghia1423**

### CODE REVIEW

File: src/main/java/io/hardingadonis/saledock/controller/others/login/ForgotPasswordServlet.java [hunk 1]
```diff
@@ -1,20 +1,108 @@
 package io.hardingadonis.saledock.controller.others.login;
 
+import io.hardingadonis.saledock.model.Employee;
+import io.hardingadonis.saledock.utils.Hash;
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/controller/others/login/ForgotPasswordServlet.java]: Use `import io.hardingadonis.saledock.utils.*;`
Use `import io.hardingadonis.saledock.model.*;`

File: src/main/java/io/hardingadonis/saledock/controller/others/login/ForgotPasswordServlet.java [hunk 2]
```diff
@@ -1,20 +1,108 @@
...
+import io.hardingadonis.saledock.utils.OtpUltil;
+import io.hardingadonis.saledock.utils.SendEmailUtil;
+import io.hardingadonis.saledock.utils.SessionUtil;
+import io.hardingadonis.saledock.utils.Singleton;
 import java.io.*;
 import jakarta.servlet.*;
 import jakarta.servlet.annotation.*;
 import jakarta.servlet.http.*;
+import java.util.Optional;
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/controller/others/login/ForgotPasswordServlet.java]: Use `import java.util.*;`

File: src/main/webapp/view/jsp/others/login/forgot-password.jsp [hunk 3]
```diff
@@ -0,0 +1,76 @@
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+
+<!DOCTYPE html>
+<html data-bs-theme="light" lang="en">
+
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0, shrink-to-fit=no">
+        <title>Forgotten Password - Sale Dock</title>
```
Comment (hardingadonis) [inline src/main/webapp/view/jsp/others/login/forgot-password.jsp]: Use vietnamese for tittle

File: src/main/webapp/view/jsp/others/login/new-password.jsp [hunk 4]
```diff
@@ -0,0 +1,95 @@
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+
+<!DOCTYPE html>
+<html data-bs-theme="light" lang="en">
+
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0, shrink-to-fit=no">
+        <title>Forgotten Password - Sale Dock</title>
```
Comment (hardingadonis) [inline src/main/webapp/view/jsp/others/login/new-password.jsp]: Use vietnamese for tittle

File: src/main/webapp/view/jsp/others/login/otp.jsp [hunk 5]
```diff
@@ -0,0 +1,75 @@
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+
+<!DOCTYPE html>
+<html data-bs-theme="light" lang="en">
+
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0, shrink-to-fit=no">
+        <title>Forgotten Password - Sale Dock</title>
```
Comment (hardingadonis) [inline src/main/webapp/view/jsp/others/login/otp.jsp]: Use vietnamese for tittle

**Comment by htnghia1423** [comment]
@hardingadonis done. Check lại xem thử 🐸

File: src/main/java/io/hardingadonis/saledock/utils/SendEmailUtil.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
+package io.hardingadonis.saledock.utils;
+
+import jakarta.mail.Authenticator;
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/utils/SendEmailUtil.java]: Use `import jakarta.mail.*;`

File: src/main/java/io/hardingadonis/saledock/utils/SendEmailUtil.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+import jakarta.mail.Message;
+import jakarta.mail.MessagingException;
+import jakarta.mail.PasswordAuthentication;
+import jakarta.mail.Session;
+import jakarta.mail.Transport;
+import jakarta.mail.internet.InternetAddress;
+import jakarta.mail.internet.MimeMessage;
+
+import java.util.Date;
```
Comment (hardingadonis) [inline src/main/java/io/hardingadonis/saledock/utils/SendEmailUtil.java]: Use `import java.util.*;`

**Review by hardingadonis — APPROVED** [review]

**Review by yuhtnguyen — APPROVED** [review]

**Review by bakaqc — APPROVED** [review]

**Pull request merged**